src/composed/cleartext.rs

```diff
@@ -305,18 +305,35 @@ fn dash_escape(text: &str) -> String {
     out
 }
 
-/// Undo dash escaping of `text`.
+/// Undo dash escaping of `text`, and trim space/tabs at the end of lines.
 ///
-/// This implementation is implicitly agnostic between "\n" and "\r\n" line endings.
-fn dash_unescape(text: &str) -> String {
+/// This implementation can handle both "\n" and "\r\n" line endings.
+fn dash_unescape_and_trim(text: &str) -> String {
```

Comment:
it would be good to have unit test for this behaviour, just testing this function

Reply:
good point. i've added a unit test for just the unescaper.